src/v2/cookbook/how-to-use-the-vue-plugin-api.md

```diff
@@ -0,0 +1,214 @@
+
+Usually we would use Vuex like this in a Vue component:
+
+```js
+import { mapState, mapMutations } from 'vuex'
+
+export default {
+  computed: {
+    ...mapState({
+      user: (state) => state.user,
+    }),
+    ...mapMutations(['UPDATE_USER']),
+  },
+}
+```
+
+Our abstraction would look something like this:
+
+```js
+export default {
+  vuex: {
+    state: {
+      user: (state) => state.user,
+    },
+    mutations: ['UPDATE_USER'],
+  },
+}
+```
+
+## Let's build the plugin
+
+### Setting up the plugin
+
+Before we can get into the logic of the plugin, we first need to setup the plugin in a `.js` file
+
+```js
+import Vue from 'vue'
+
+const plugin = {
+  install(LocalVue) {
+    
+  }
+}
+
+Vue.use(plugin)
+```
+
+You might have come across `Vue.use()` before, the method is responsible for registering plugins and it takes an object with a method named `install` which gets called.
+
+The `install` hook exposes two parameters:
+
+* First parameter is the Vue constructor
+* Second parameter is the options that you pass to the plugin
+
+Next step is to add a global mixin in the `install` hook, we can use the `beforeCreate` lifecycle hook to add/modify the Vue component instance
+
+```js
+import Vue from 'vue'
+
+const plugin = {
+  install(LocalVue) {
+    LocalVue.mixin({
+      beforeCreate() {
+        this // we can access the component instance
+      }
+    })
+  }
+}
+
+Vue.use(plugin)
+```
+
+### Writing the plugin logic
+
+Now that we have all the plugin boilerplate setup, let's have a look at actually taking our abstraction and converting it to something that Vue understands.
+
+Every Vue instance has a property called `$options` which contains a reference to all the options exported in the Vue component, you can read more about the [Options API](https://vuejs.org/v2/api/?#Options-Data).
+
+#### Step 1 - Access vuex from options
+
+```js
+import Vue from 'vue'
+
+const plugin = {
+  install(LocalVue) {
+    LocalVue.mixin({
+      beforeCreate() {
+        const { 
+          vuex = {},
+          methods = {},
+          computed = {},
+        } = this.$options
+      }
+    })
+  }
+}
+
+Vue.use(plugin)
+```
+
+Here we are destructuring the `vuex`, `methods` and `computed` properties from the `$options` object and if it does not exist we can set a default of an empty object.
+
+#### Step 2 - Access props from vuex
+
+```js
+import Vue from 'vue'
+
+const plugin = {
+  install(LocalVue) {
+    LocalVue.mixin({
+      beforeCreate() {
+        const { 
+          vuex = {},
+          methods = {},
+          computed = {},
+        } = this.$options
+
+        const {
+          state = {},
+          getters = {},
+          mutations = {},
+          actions = {},
+        } = vuex
+      }
+    })
+  }
+}
+
+Vue.use(plugin)
+```
+
+Next step is to destructure the `state`, `getters`, `mutations` and `actions` from the `vuex` property and set default objects as well.
+
+#### Step 3 - Map vuex
+
+```js
```

Comment:
So, this is all good, but there's no real use case you're covering here. Without a real use case, it's tough for the reader to understand how this is useful. I like what you have- no need to change that, but we do ask for an actual example to bring the cookbook to life and out of just a conceptual model.

Reply:
Cool thanks, so you mean a demo of something correct?

Reply:
I think I have a really real-world use case that I can add to the cookbook, I will update it :)